### PR TITLE
Update transaction handling and IT

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**
* Group commands in transaction IT.
* Ease transaction IT updating and fixing.
* Rename keys and values used in transaction IT to ease debugging.
* Rename `CommandManager` transaction related methods.
* Add number of commands missed in transaction IT.

### CODE REVIEW

File: java/integTest/src/test/java/glide/TransactionTestUtilities.java [hunk 1]
```diff
@@ -10,81 +10,81 @@
...
+    private static final String stringKey1 = "{key}" + UUID.randomUUID();
+    private static final String stringKey2 = "{key}" + UUID.randomUUID();
+    private static final String stringKey3 = "{key}" + UUID.randomUUID();
+    private static final String hashKey1 = "{key}" + UUID.randomUUID();
+    private static final String listKey1 = "{key}" + UUID.randomUUID();
+    private static final String listKey2 = "{key}" + UUID.randomUUID();
+    private static final String setKey1 = "{key}" + UUID.randomUUID();
+    private static final String zsetKey1 = "{key}" + UUID.randomUUID();
+    private static final String value1 = "value1-" + UUID.randomUUID();
+    private static final String value2 = "value2-" + UUID.randomUUID();
```
Comment (SanHalacogluImproving) [inline java/integTest/src/test/java/glide/TransactionTestUtilities.java:22]: Nit: Why not change the keyNames from {key} to more descriptive ones like "value1-" ...

Comment (Yury-Fridlyand) [reply java/integTest/src/test/java/glide/TransactionTestUtilities.java:22]: Fixed thanks

File: java/integTest/src/test/java/glide/TransactionTestUtilities.java [hunk 2]
```diff
@@ -10,81 +10,81 @@
...
-    private static final String key6 = "{key}" + UUID.randomUUID();
-    private static final String key7 = "{key}" + UUID.randomUUID();
-    private static final String key8 = "{key}" + UUID.randomUUID();
-    private static final String value1 = UUID.randomUUID().toString();
-    private static final String value2 = UUID.randomUUID().toString();
-    private static final String value3 = UUID.randomUUID().toString();
-    private static final String field1 = UUID.randomUUID().toString();
-    private static final String field2 = UUID.randomUUID().toString();
-    private static final String field3 = UUID.randomUUID().toString();
+    private static final String stringKey1 = "stringKey1-" + UUID.randomUUID();
```
Comment (acarbonetto) [inline java/integTest/src/test/java/glide/TransactionTestUtilities.java:13]: As discussed during our meeting, this can cause keys to span multiple slots and nodes.  A single transaction must be executed by a single node, so we may have problems if you rename the keys.  
i'd suggest reverting the name change.

Comment (Yury-Fridlyand) [reply java/integTest/src/test/java/glide/TransactionTestUtilities.java]: I found a way how to bypass that

File: java/client/src/main/java/glide/managers/CommandManager.java [hunk 3]
```diff
@@ -177,7 +163,7 @@ protected RedisRequest.Builder prepareRedisRequest(Transaction transaction) {
      *     adding a callback id.
      */
     protected RedisRequest.Builder prepareRedisRequest(
-            ClusterTransaction transaction, Optional<Route> route) {
+            BaseTransaction<?> transaction, Optional<Route> route) {
```
Comment (acarbonetto) [inline java/client/src/main/java/glide/managers/CommandManager.java:166]: There was a reason we split this command into Cluster and Standalone transactions.  The standalone doesn't take a route, so there was no reason to create the Optional.empty().  

I would suggest reverting the refactor.  I don't see the benefit of this.

File: java/integTest/src/test/java/glide/TransactionTestUtilities.java [hunk 4]
```diff
@@ -10,81 +10,81 @@
...
-    private static final String key6 = "{key}" + UUID.randomUUID();
-    private static final String key7 = "{key}" + UUID.randomUUID();
-    private static final String key8 = "{key}" + UUID.randomUUID();
-    private static final String value1 = UUID.randomUUID().toString();
-    private static final String value2 = UUID.randomUUID().toString();
-    private static final String value3 = UUID.randomUUID().toString();
-    private static final String field1 = UUID.randomUUID().toString();
-    private static final String field2 = UUID.randomUUID().toString();
-    private static final String field3 = UUID.randomUUID().toString();
+    private static final String stringKey1 = "{test}:stringKey1-" + UUID.randomUUID();
```
Comment (acarbonetto) [inline java/integTest/src/test/java/glide/TransactionTestUtilities.java:13]: I suggest only renaming the variable.  I don't see the benefit of renaming the key.

File: java/integTest/src/test/java/glide/cluster/ClusterTransactionTests.java [hunk 5]
```diff
@@ -17,12 +17,13 @@
...
 import glide.api.models.configuration.RedisClusterClientConfiguration;
 import java.util.Arrays;
-import java.util.concurrent.TimeUnit;
 import lombok.SneakyThrows;
 import org.junit.jupiter.api.AfterAll;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
 
+@Timeout(20)
```
Comment (acarbonetto) [inline java/integTest/src/test/java/glide/cluster/ClusterTransactionTests.java:26]: make this a separate PR please

Comment (Yury-Fridlyand) [reply java/integTest/src/test/java/glide/cluster/ClusterTransactionTests.java]: Moved to  #170

File: java/client/src/main/java/glide/managers/CommandManager.java (hunk 3 above)
Comment (Yury-Fridlyand) [reply java/client/src/main/java/glide/managers/CommandManager.java]: Reverted

File: java/integTest/src/test/java/glide/TransactionTestUtilities.java (hunk 4 above)
Comment (Yury-Fridlyand) [reply java/integTest/src/test/java/glide/TransactionTestUtilities.java:13]: It helped for post-mortem debugging the test - I kept the cluster running for querying them.

**Comment by Yury-Fridlyand** [comment]
Round

**Review by acarbonetto — APPROVED** [review]

**Review by jonathanl-bq — APPROVED** [review]
